### ISSUE TITLE
elasticsearch_hot_thread_cpu_usage metric is not avail

### ISSUE THREAD

**Issue by jainpratik163** [prometheus-community/elasticsearch_exporter#949]
i have installed es exporter for getting es data and i am trying to find elasticsearch_hot_thread_cpu_usage metric but not able to find such metric also.

**Comment by ulascansenturk** [comment]
I'd like to work on this, could you assign it to me? @SuperQ @sysadmind

**Comment by hiSandog** [comment]
What exact value should `elasticsearch_hot_thread_cpu_usage` represent: the top thread's reported CPU percentage, a sum across returned hot threads, or per-node/per-thread series? The hot-threads API is diagnostic text and can be relatively expensive, so scrape-time parsing needs clear opt-in semantics and bounded cardinality. A sample Elasticsearch response/version plus desired labels would make the metric contract implementable and prevent unstable thread names from exploding series count.

**Comment by SuperQ** [comment]
Yes, it's typically better to report CPU usage as a counter. For some services where I need thread details I use a [process exporter](https://github.com/ncabatoff/process-exporter). This uses OS thread views and avoids hitting the workload directly. Perhaps there's a JVM way to do something similar?

**Comment by f9n** [comment]
In our setup we often run against managed Elasticsearch (Elastic Cloud, ECK, Aws Opensearch etc.) where we don't control the underlying nodes or JVM/process visibility. In those cases a sidecar/process exporter on ES nodes isn't an option, so exposing hot-thread CPU via the exporter/API is the practical path for us.

+1 on making this opt-in and disabled by default (`--collector.nodes_hot_threads`). The hot_threads endpoint is diagnostic and relatively expensive, so teams that need it can enable it explicitly.

For semantics/cardinality: per-node series for the returned hot threads is sufficient for troubleshooting. It would also help if the collector calls the API with `ignore_idle_threads=true` (and a sensible threads limit) so idle threads don't add noise or unnecessary series.

Thanks for efforts ❤️
